channel-sdk/src/androidTest/java/com/zello/channel/sdk/SessionTests.kt

```diff
@@ -0,0 +1,1044 @@
+package com.zello.channel.sdk
+
+import android.graphics.Bitmap
+import android.location.Criteria
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.zello.channel.sdk.image.Dimensions
+import com.zello.channel.sdk.image.ImageMessageManager
+import com.zello.channel.sdk.image.ImageMessageManagerListener
+import com.zello.channel.sdk.image.ImageTag
+import com.zello.channel.sdk.image.TestImageUtils
+import com.zello.channel.sdk.location.LocationManager
+import com.zello.channel.sdk.platform.AudioReceiver
+import com.zello.channel.sdk.platform.AudioReceiverEvents
+import com.zello.channel.sdk.platform.AudioSource
+import com.zello.channel.sdk.platform.AudioSourceEvents
+import com.zello.channel.sdk.platform.Decoder
+import com.zello.channel.sdk.platform.DecoderListener
+import com.zello.channel.sdk.platform.Encoder
+import com.zello.channel.sdk.platform.EncoderListener
+import com.zello.channel.sdk.platform.PlayerListener
+import com.zello.channel.sdk.platform.hexString
+import com.zello.channel.sdk.transport.Transport
+import com.zello.channel.sdk.transport.TransportFactory
+import org.json.JSONObject
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.ByteArrayOutputStream
+import java.util.Arrays
+
+//region Mocks
+
+internal class TestAudioSource: AudioSource {
+	override val level: Int = 0
+
+	override fun prepare(sampleRate: Int, bufferSampleCount: Int, levelMeter: Boolean, noiseSuppression: Boolean, useAGC: Boolean): Boolean = false
+
+	override fun start() { }
+
+	override fun stop() { }
+}
+
+internal class TestEncoder: Encoder {
```

Comment:
Maybe we can add mockito in future to help with mocking of some of the bigger but mostly empty objects.

Reply:
It's a good idea. I didn't want to dig into it just now.